lib/course.js

```diff
@@ -263,12 +260,41 @@ const applyGlobalOptions = (result, opts, relativePath, parserVersion) => ({
 //
 
 
-module.exports = (dir, { Topic }, opts = {}) => {
-  const readme = path.join(dir, common.getReadmeFileName(opts.locale));
-  const relativePath = path.relative(process.cwd(), dir);
-  const { slug } = common.parseDirname(readme);
+module.exports = async (dirOrPath, { Topic }, opts = {}) => {
```

Comment:
The variable name `dirOrPath` is really ugly. Can you help me identify a better name?

Reply:
I was thinking of location which might refer to a location in the File System or a location on the web (a URL) but not sure if it is okay.

Reply:
maybe just "location"